src/config.c

```diff
@@ -511,7 +511,14 @@ static const char *default_video_device(void)
 #elif defined (WIN32)
 	return "dshow,nil";
 #else
-	return "v4l2,/dev/video0";
+	re_hprintf(pf, "%s", "v4l2,");
+	const struct vidsrc *vs = vidsrc_find(baresip_vidsrcl(), "v4l2");
+	if(vs) {
+		return re_hprintf(pf, "%s", mediadev_get_default(&vs->dev_list)->name);
+	}
+	else {
+		return re_hprintf(pf, "%s", "/dev/video0");
+	}
```

Comment:
this will break the cases for the other `#ifdef`s since the signature of the
function is changed.

I think this part is a nice improvement, but if you want we can also leave it to another patch.

the printing of the default device should be shared for all platforms, something like:

1. find the default vidsrc module for the current platform
2. find the default device name for that module.